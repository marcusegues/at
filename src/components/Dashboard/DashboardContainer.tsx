import React, { Component, Dispatch } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { REQUEST_FETCH_CURRENCY_PAIRS } from '../../actions/currencyPairs';
import { CurrencyPairsActionTypes } from '../../actions/currencyPairs/types';
import { requestFetchOrdersAction } from '../../actions/orders';
import { OrdersActionTypes } from '../../actions/orders/types';
import { OrdersTableContainer } from '../OrdersTable/OrdersTableContainer';

interface DispatchProps {
  onFetchCurrencyPairs: () => void;
  onFetchOrders: () => void;
}

type Props = DispatchProps;

export class DashBoardContainerInner extends Component<Props> {
  public state = {};

  public async componentDidMount() {
    this.props.onFetchCurrencyPairs();
    this.props.onFetchOrders();
  }

  public render() {
    return (
      <Container>
        <OrdersTableContainer />
      </Container>
    );
  }
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 20px;
  margin-right: 20px;
  width: 620px;
`;

const mapDispatchToProps = (dispatch: Dispatch<CurrencyPairsActionTypes | OrdersActionTypes>) => ({
  onFetchCurrencyPairs: () => dispatch({ type: REQUEST_FETCH_CURRENCY_PAIRS }),
  onFetchOrders: () => dispatch(requestFetchOrdersAction()),
});

export const DashboardContainer = connect(
  null,
  mapDispatchToProps
)(DashBoardContainerInner);
