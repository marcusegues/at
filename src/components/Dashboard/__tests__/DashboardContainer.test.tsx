import { shallow } from 'enzyme';
import React from 'react';
import { DashboardContainer } from '../DashboardContainer';
import { DashBoardContainerInner } from '../DashboardContainer';

describe('DashboardContainer', () => {
  const onFetchCurrencyPairsMock = jest.fn();
  const onFetchOrdersMock = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  // This test should actually be implemented using Enzyme's mount or react-test-renderer create, however
  // ag-grid presents some issues (tracked in their repo) that need workarounds, and I am not including the latter here
  // the below test is not ideal because I am mocking what the connected component would get from react-redux, which
  // is defined in the component itself: this represents a decoupling between implementation and tests which should
  // definitely be solved when the ag-grid workarounds are in place
  test('calls passed in actions', () => {
    shallow(
      <DashBoardContainerInner
        onFetchCurrencyPairs={onFetchCurrencyPairsMock}
        onFetchOrders={onFetchOrdersMock}
      />
    );

    expect(onFetchCurrencyPairsMock).toHaveBeenCalled();
    expect(onFetchOrdersMock).toHaveBeenCalled();
  });

  test('calls passed in actions', () => {
    const component = shallow(
      <DashBoardContainerInner
        onFetchCurrencyPairs={onFetchCurrencyPairsMock}
        onFetchOrders={onFetchOrdersMock}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
