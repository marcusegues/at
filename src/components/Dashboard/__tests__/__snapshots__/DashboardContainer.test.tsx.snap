// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardContainer calls passed in actions 1`] = `
<styled.div>
  <Connect(OrdersTableContainerInner) />
</styled.div>
`;
